src/features/search/components/stockList.tsx

```diff
@@ -63,21 +63,20 @@ type _Stock = {
 }
 
 type StockListProps = {
-    setQuery: Dispatch<SetStateAction<string>>,
     inputSelect: RefObject<HTMLInputElement>,
     setSelectedStock: Dispatch<SetStateAction<string[]>>,
     dropSelect: RefObject<HTMLDivElement>,
 }
 
-export const StockList = memo<StockListProps>(({setQuery, inputSelect, setSelectedStock, dropSelect}) => {
+export const StockList = memo<StockListProps>(({inputSelect, setSelectedStock, dropSelect}) => {
 
     const dispatch = useDispatch()
     const stockList = useSelector((state: AppState) => state.search.stockList)
 
     const onStockClick = (stock: _Stock) => {
         const stockSymbol = stock.symbol
         const stockName = stock.name
-        setQuery(`${stockName} (${stockSymbol})`)
+        dispatch(updateQuery(`${stockName} (${stockSymbol})`))
```

Comment:
Q: If I understand correctly, this action is supposed to send the search input on every key press to the server. Can you explain why you need to send the stock name as well?

Reply:
So here query is being updated for label overlay purposes.

The overlay has to = query in order to match match on/off of focus.